File: DSA/DivyanshKumarSingh_590012175/Day_17_Question_3.py
n = int(input())

def sq(n):
    l, r, ans = 0, n, 0
    while l <= r:
        m = (l + r) // 2
        if m * m <= n:
            ans = m
            l = m + 1
        else:
            r = m - 1
    return ans

print(sq(n))

File: DSA/DivyanshKumarSingh_590012175/Day_17_Question_3.c
#include <stdio.h>

int sq(int n)
{
    int l = 0, r = n, ans = 0;
    while (l <= r)
    {
        int m = (l + r) / 2;
        if (m * m <= n)
        {
            ans = m;
            l = m + 1;
        }
        else
        {
            r = m - 1;
        }
    }
    return ans;
}

int main()
{
    int n;
    scanf("%d", &n);
    printf("%d\n", sq(n));
    return 0;
}
